src/particles/diagnostics/DiagnosticOutput.cpp

```diff
@@ -26,52 +27,66 @@ namespace impactx::diagnostics
         tmp.copyParticles(pc, local);
 
         // write file header
-        amrex::PrintToFile(file_name) << "x y t px py pt\n";
-
-        // loop over refinement levels
-        int const nLevel = tmp.finestLevel();
-        for (int lev = 0; lev <= nLevel; ++lev)
+        if (amrex::ParallelDescriptor::IOProcessor())
         {
-            // loop over all particle boxes
-            using ParIt = typename decltype(tmp)::ParConstIterType;
-            for (ParIt pti(tmp, lev); pti.isValid(); ++pti) {
-                const int np = pti.numParticles();
+            amrex::PrintToFile(file_name) << "x y t px py pt\n";
+        }
+
+        const int MyProc = amrex::ParallelDescriptor::MyProc();
+
+        for (int proc = 0; proc < amrex::ParallelDescriptor::NProcs(); proc++) {
+            // note: this is the same logic as in amrex::ParticleContainer::WriteAsciiFile
+            //   this will obviously not scale and also has the assumption
```

Comment:
Yes, the ascii format is meant for small tests and should not be used at scale.

Reply:
Correct, also here. Maybe I should add a note in the API interface of the function that this is temporary.